source/vision/visionHandler.py

```diff
@@ -48,18 +49,73 @@ def getProviderClass(
 			raise initialException
 
 
+def getProviderList(
+		onlyStartable: bool = True
+) -> List[providerInfo.ProviderInfo]:
+	"""Gets a list of available vision enhancement provider information
+	@param onlyStartable: excludes all providers for which the check method returns C{False}.
+	@return: List of available providers
+	"""
+	providerList = []
+	for loader, moduleName, isPkg in pkgutil.iter_modules(visionEnhancementProviders.__path__):
+		if moduleName.startswith('_'):
+			continue
+		try:
+			provider = getProviderClass(moduleName)
+		except Exception:
+			# Purposely catch everything.
+			# A provider can raise whatever exception it likes,
+			# therefore it is unknown what to expect.
+			log.error(
+				f"Error while importing vision enhancement provider module {moduleName}",
+				exc_info=True
+			)
+			continue
+		try:
+			if not onlyStartable or provider.canStart():
+				providerSettings = provider.getSettings()
+				providerList.append(
+					providerInfo.ProviderInfo(
+						providerId=providerSettings.getId(),
+						moduleName=moduleName,
+						translatedName=providerSettings.getTranslatedName(),
+						providerClass=provider
+					)
+				)
+			else:
+				log.debugWarning(
+					f"Excluding Vision enhancement provider module {moduleName} which is unable to start"
+				)
+		except Exception:
+			# Purposely catch everything else as we don't want one failing provider
+			# make it impossible to list all the others.
+			log.error("", exc_info=True)
+	# Sort the providers alphabetically by name.
+	providerList.sort(key=lambda info: info.translatedName.lower())
+	return providerList
+
+
+def getProviderInfo(providerId: providerInfo.ProviderIdT) -> Optional[providerInfo.ProviderInfo]:
+	# This mechanism of getting the provider list and looking it up is particularly inefficient, but, before
+	# refactoring, confirm that getProviderList is / isn't cached.

```

Comment:
I think this approach is fine for now. But it might be worth investigating whether it can be made more efficient. I don't really like the idea of generating this list so often. One benefit though is that it would respond to newly added files. If the list is cached by the autoProperty code, then it might not require any work.